modules/amazon/index.js

```diff
@@ -0,0 +1,60 @@
+var amazon = require('amazon-product-api');
+var amazonClient = null;
+
+module.exports.init = function(bot) {
+  bot.getConfig("amazon.json", function(err, conf) {
+    if(err) bot.say("Unable to load amazon module: " + err);
+    try {
+      amazonClient = new amazon.createClient(conf);
+    } catch(ex) {
+      bot.say("Error loading amazon library: " + ex);
+    }
+  });
+};
+
+var extractAsin = function(url) {
+  var asin;
+  var amazonRegex = /(B[0-9]{2}[0-9A-Z]{7}|[0-9]{9}(?:X|[0-9]))/;
+  match = amazonRegex.exec(url);
+  if(match && match[1])
+    asin = match[0];
+  return asin;
+};
+
+var generateErrorMessage = function(error) {
+  var err = error[0].Error[0];
```

Comment:
Y'know, if you checked to make sure that these lists were nonempty, you could sneak in an `"Error: Error does not contain error"` message =D

Reply:
(this is mostly a non-serious suggestion because it sounds like the amazon API is just bad)